lib/PuppeteerSharp.Tests/Page/CloseTests.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Threading.Tasks;
+using Xunit;
+
+namespace PuppeteerSharp.Tests.Page
+{
+    [Collection("PuppeteerLoaderFixture collection")]
+    public class CloseTests : PuppeteerBaseTest
+    {
+        [Fact]
+        public async Task ShouldRejectAllPromisesWhenPageIsClosed()
+        {
+            var newPage = await Browser.NewPageAsync();
+            var neverResolves = newPage.EvaluateFunctionAsync("() => new Promise(r => {})");
+            
+            await newPage.CloseAsync();
```

Comment:
One thing I noticed on Puppeteer is that they are not awaiting the close
https://github.com/GoogleChrome/puppeteer/blob/v1.0.0/test/test.js#L325